hostinfo/hostinfo.go

```diff
@@ -321,6 +326,40 @@ func inDockerDesktop() bool {
 	return false
 }
 
+// goArchVar returns the GOARM or GOAMD64 etc value that the binary was built
+// with.
+func goArchVar() string {
```

Comment:
If a GOARM64 or GORISCV is ever added I don't think we have a way to automatically detect it, I don't see a way to check https://go.dev/src/internal/buildcfg/cfg.go in a CI test or something.

Reply:
Okay, I rewrote it to be future-proof, as long as the future follows the conventions thus far. Not that we really care about anything except GOARM, which is how this all started 😅